### ISSUE TITLE
Rule S2681: Expand implementation for additional patterns

### ISSUE THREAD

**Issue by ganncamp** [SonarSource/sonar-dotnet#194]
This rule implementation should also catch the following patterns:
```
if(condition) firstActionInBlock(); secondAction();  // Noncompliant; secondAction executed unconditionally

if(condition) firstActionInBlock();  // Noncompliant
  secondAction();  // Executed unconditionally
```
